packages/core/src/browser/label-parser.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+import { injectable } from "inversify";
+
+export interface LabelIcon {
+    name: string;
+    animation?: string;
+}
+
+export namespace LabelIcon {
+    export function is(val: object): val is LabelIcon {
+        return 'name' in val;
+    }
+}
+
+export type LabelPart = string | LabelIcon;
+
+@injectable()
+export class LabelParser {
```

Comment:
Could we had a comment here, I just don't get what this is doing... and understanding the code is non obvious

Reply:
Wrote it on parser method.